test/e2e/utils/fixtures.ts

```diff
@@ -4,32 +4,74 @@ import type { Browser, Page } from '@playwright/test';
 import type { usernamesForFixture } from './userFixtures';
 import constants from '../testConstants.json';
 
-export type UserTab = Page & {
+export type UserDetails = {
   username: string,
   password: string,
   name: string,
   email: string,
 }
 
+export type UserTab = Page & UserDetails;
+
+function setupUserDetails(obj: any, username: usernamesForFixture) {
+  obj.username = constants[`${username}Username`] ?? username;
+  obj.name = constants[`${username}Name`] ?? username;
+  obj.password = constants[`${username}Password`] ?? 'x';
+  obj.email = constants[`${username}Email`] ?? `${username}@example.com`;
+}
+
 const userTab = (username: usernamesForFixture) => async ({ browser, browserName }: { browser: Browser, browserName: string}, use: (r: UserTab) => Promise<void>) => {
   const storageState = `${browserName}-${username}-storageState.json`;
   const context = await browser.newContext({ storageState })
   const page = await context.newPage();
   const tab = page as UserTab;
-  tab.username = constants[`${username}Username`] ?? username;
-  tab.name = constants[`${username}Name`] ?? username;
-  tab.password = constants[`${username}Password`] ?? 'x';
-  tab.email = constants[`${username}Email`] ?? `${username}@example.com`;
+  setupUserDetails(tab, username);
   await use(tab);
-  await tab.close();
-  await context.close();
 }
 
 // Extend basic test by providing a "todoPage" fixture.
```

Comment:
we need a different comment here to explain the purpose of the enhanced test fixture functionality